bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java

```diff
@@ -148,7 +148,7 @@ public void fillContextMenu(IMenuManager menu) {
 		// Add the 'refresh' item if any selection is either (a) an open project, or (b)
                	// a non-project selection (so the 'refresh' item is not shown if all selections
                	// are closed projects)
-               	if (hasOpenProjects || !isProjectSelection) {
+		if (hasOpenProjects || projects.size() < selection.size()) {
```

Comment:
please add explanation why `projects.size() < selection.size()` i sused here. That is not self-explaining

Reply:
maybe introduce a variable like selectionContainsNonProject